photon-client/src/main/scala/com/linkedin/photon/ml/io/scopt/ScoptParserHelpers.scala

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.io.scopt
+
+import java.util.StringJoiner
+
+import scala.util.Try
+import scala.collection.mutable
+
+import com.linkedin.photon.ml.Types.{CoordinateId, FeatureShardId}
+import com.linkedin.photon.ml.data.{FixedEffectDataConfiguration, InputColumnsNames, RandomEffectDataConfiguration}
+import com.linkedin.photon.ml.io.{CoordinateConfiguration, FeatureShardConfiguration, FixedEffectCoordinateConfiguration, RandomEffectCoordinateConfiguration}
+import com.linkedin.photon.ml.optimization._
+import com.linkedin.photon.ml.optimization.RegularizationType._
+import com.linkedin.photon.ml.optimization.game.{FixedEffectOptimizationConfiguration, GLMOptimizationConfiguration, RandomEffectOptimizationConfiguration}
+import com.linkedin.photon.ml.projector.IdentityProjection
+import com.linkedin.photon.ml.util.{DoubleRange, Logging, PhotonLogger}
+
+/**
+ * Helper values/functions for parsing Scopt parameters.
+ */
+object ScoptParserHelpers extends Logging {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

Helper functions for non-trivial Scopt parsing and printing. This file consolidates all of the info that used to be stored in individual classes (e.g. `OptimizerConfig`, `FixedEffectDataConfiguration`, etc.).